### DISCUSSION TITLE
Chakra-ui with Server Side Components?

### DISCUSSION THREAD

**Discussion by KeisukeNagakawa** [chakra-ui/chakra-ui#6941] [General]
I love chakra-ui and would like to use it in `Server Side Component` in nextjs, because using Chakra only in Client Side Component is not flexible. Do you have any plan for enabling this functionality? Thanks!

**Comment by Southclaws** [comment]
I am wondering the same thing, I wanted to try the new SEO metadata APIs but it seems impossible as they only work with server side components and chakra only works with client side components. https://beta.nextjs.org/docs/api-reference/metadata

**Comment by Ali-Aref** [comment]
any updates on this ?

**Comment by Southclaws** [comment]
Segun's post is a good summary of what the plan is: https://www.adebayosegun.com/blog/chakra-panda-ark-whats-the-plan

As for myself, I've settled on using Panda for 80% of my layout needs and content now as it's fully server-side rendered with statically generated CSS and then for the interactive parts, I use Chakra. This gives me the best of both worlds where speed and SEO matters for the core content, then Chakra acts as a progressive enhancement on top of this for highly interactive areas of an application.

What's also worth noting, which I didn't quite understand initially, is that when using Next.js the first render is all on the server anyway - even for "use client" files, it essentially renders whatever it can then hydrates on the client and carries on in a progressively enhanced way. The only time you'll run into issues and need explicit `typeof window === undefined` style checks is when you're actively using browser APIs in your components. Then you just treat it as you always have with SSR - render a skeleton or a spinner or whatever on the server, pick it up on the client.

Edit: I also wrote a small summary of steps required to switch from Chakra to Panda -> https://southcla.ws/how-to-migrate-from-chakra-ui-to-panda-css

**Comment by Sam152** [comment]
Hm, reading between the lines on the blog post, if Chakra eventually transitions to Panda, the components won't require client-side hydration and will appear styled after being delivered by the main document request to the server. Is that correct?

I feel like that's an important part of the value proposition of integrating with Next and other SSR approaches, since you want to avoid blocking properly rendered content behind HTTP request waterfalls.

Hopefully that's the direction things are going in, then Chakra can be deployed for apps that are optimising for healthy core web vitals.

**Comment by KamranTaghaddos** [comment]
> I love chakra-ui and would like to use it in `Server Side Component` in nextjs, because using Chakra only in Client Side Component is not flexible. Do you have any plan for enabling this functionality? Thanks!

This is the chakra UI doc text: "Next.js 13 introduced a new app/ directory / folder structure. By default it uses Server Components. However, Chakra UI only works in client-side components". So, as you see the chakra doc says that all chakra components are client-side component.

Here is the reference: [https://v2.chakra-ui.com/getting-started/nextjs-app-guide](https://v2.chakra-ui.com/getting-started/nextjs-app-guide)
